### PR TITLE
add new setup commands

### PR DESCRIPTION
**Pull request by ImYenil**
Added new command

/bw setspawnprotection <int>
/bw setshopprotection <int>
/bw setupgradesprotection <int>
/bw setgeneratorprotection <int>
/bw setislandradius <int>
/bw setusebedhologram <boolean>
/bw setdisplayname <name>

### CODE REVIEW

**Comment by tomkeuper** [comment]
dont merge to master pls

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetGeneratorProtection.java
Comment (tomkeuper) [inline bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetGeneratorProtection.java:70]: Im quite sure you copied this from the setMaxBuildHeight. You should probably make it null so people can put their own value there without these abnormally height numbers

**Comment by tomkeuper** [comment]
All is good just change the commands that have a tabcomplete array copied from setMaxBuildHeight

**Comment by tomkeuper** [comment]
After this merge you might need to re fork the project because for some reason you keep having the workflow update commits, I squashed them before

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetGeneratorProtection.java
Comment (tomkeuper) [inline bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetGeneratorProtection.java]: Pleace correct the value to be appropriate to the setting or remove the auto fill values

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetIslandRadius.java
Comment (tomkeuper) [inline bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetIslandRadius.java]: Pleace correct the value to be appropriate to the setting or remove the auto fill values

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetShopProtection.java
Comment (tomkeuper) [inline bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetShopProtection.java]: Pleace correct the value to be appropriate to the setting or remove the auto fill values

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetSpawnProtection.java
Comment (tomkeuper) [inline bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetSpawnProtection.java]: Pleace correct the value to be appropriate to the setting or remove the auto fill values

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetUpgradesProtection.java
Comment (tomkeuper) [inline bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetUpgradesProtection.java]: Pleace correct the value to be appropriate to the setting or remove the auto fill values

**Review by tomkeuper — CHANGES_REQUESTED** [review]

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetUpgradesProtection.java
Comment (ImYenil) [reply bedwars-plugin/src/main/java/com/andrei1058/bedwars/commands/bedwars/subcmds/sensitive/setup/SetUpgradesProtection.java]: > Pleace correct the value to be appropriate to the setting or remove the auto fill values

all solved

**Review by porudev — COMMENTED** [review]
i can only get the set display name, the rest i don't get the point can you explain?

**Comment by tomkeuper** [comment]
> i can only get the set display name, the rest i don't get the point can you explain?

I agree with you a little bit but maybe its better to have the option to change certain variables as this doesnt add/change to normal gameplay and doesnt hurt performance. 
And ofcourse the options are not mandatory. 

@ImYenil have you tested if this actually works? I dont know when the save action is of arenas.

**Review by tomkeuper — APPROVED** [review]

**Comment by tomkeuper** [comment]
> > i can only get the set display name, the rest i don't get the point can you explain?
> 
> I agree with you a little bit but maybe its better to have the option to change certain variables as this doesnt add/change to normal gameplay and doesnt hurt performance. And ofcourse the options are not mandatory.
> 
> @ImYenil have you tested if this actually works? I dont know when the save action is of arenas.

@ImYenil

**Comment by ImYenil** [comment]
> > > i can only get the set display name, the rest i don't get the point can you explain?
> > 
> > 
> > I agree with you a little bit but maybe its better to have the option to change certain variables as this doesnt add/change to normal gameplay and doesnt hurt performance. And ofcourse the options are not mandatory.
> > @ImYenil have you tested if this actually works? I dont know when the save action is of arenas.
> 
> @ImYenil

I tested it works

**Review by porudev — COMMENTED** [review]
why do they return null, can you explain?

**Comment by tomkeuper** [comment]
> why do they return null, can you explain?

Null will be an empty response when a tab complete action is executed on a client. So the commands that return null will have no tab complete on the values they accept

**Pull request merged**
